QoL/Modules/PatchedBosses.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Linq;
+using HutongGames.PlayMaker;
+using MonoMod.Cil;
+using UnityEngine;
+using Vasi;
+
+namespace QoL.Modules
+{
+    public class PatchedBosses : FauxMod
+    {
+        [SerializeToSetting]
+        public static bool SporeShroomHollowKnight = true;
+
+        [SerializeToSetting]
+        public static bool NoBackrolls = false;
+
+        [SerializeToSetting]
+        public static bool SleepyCrystalGuardian = false;
+
+
+        public override void Initialize()
+        {
+            IL.ExtraDamageable.RecieveExtraDamage += AllowRecieveExtraDamage;
```

Comment:
Probably fine but does this only apply to THK? I thought I had seen ExtraDamageable in other places but I might be remembering wrong. Just want to double check

Reply:
Uhh, so in fact I think, like, every enemy has an `ExtraDamageable` component (otherwise it can't get damaged at all by spore shroom or defender's crest), but that doesn't matter here because the enemy will never become inactive and there will only be one copy of the enemy in the `DamageEffectTicker` component's `enemyList`.

THK has a bunch of colliders corresponding to, like, each attack, and each of them has the `ExtraDamageableProxy` component which can enter the `enemyList`. And, those colliders will become inactive quite frequently (I think it's only the Dash and Dash Antic colliders which I've seen leave the list rather than simply becoming inactive, other than THK itself) which leads to the massive damage.

Well anyway I don't actually know how to quickly check if there are any other objects with the `ExtraDamageableProxy` component (I think it was added after the dump in `#modding-development` was made; certainly there are no files in that dump which mention the component though), but if you prefer I can add a scene check (` && GameManager.instance.sceneName == "Room_Final_Boss_Core"`) to the checks?